tembo-operator/src/app_service/manager.rs

```diff
@@ -394,8 +395,14 @@ fn generate_deployment(
     };
     volume_mounts.push(certs_volume_mount);
 
+    let mut pod_security_context: Option<PodSecurityContext> = None;
     // Add any user provided volumes / volume mounts
     if let Some(storage) = appsvc.storage.clone() {
+        // when there are user specified volumes, we need to let kubernetes modify permissions of those volumes
+        pod_security_context = Some(PodSecurityContext {
+            fs_group: Some(65534),
```

Comment:
PodSecurityContext fsGroup

> securityContext.fsGroup (int64)
> 
> A special supplemental group that applies to all containers in a pod. Some volume types allow the Kubelet to change the ownership of that volume to be owned by the pod:
> 
> The owning GID will be the FSGroup 2. The setgid bit is set (new files created in the volume will be owned by FSGroup) 3. The permission bits are OR'd with rw-rw----
> If unset, the Kubelet will not modify the ownership and permissions of any volume. Note that this field cannot be set when spec.os.name is windows.

https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context

Reply:
It is unnecessary to have both pod security policy and container security policy, we could maybe just define this setting inside one or the other. This comment has no security benefit it is only a question of organization.

Reply:
fsGroup isn't available on the container from what I can tell, so we'd need to move it all up to the pod I think...

Reply:
In this case I think good to go as-is, since that's nicer from the perspective of changing less.